onnxruntime/core/providers/webnn/builders/impl/gemm_op_builder.cc

```diff
@@ -233,26 +233,18 @@ bool GemmOpBuilder::HasSupportedInputsImpl(const Node& node, const emscripten::v
     return false;
   }
 
-  std::string webnn_op_type;
-  if (!GetWebNNOpType(op_type, webnn_op_type))
-    return false;
-
-  if (!IsSupportedDataType(input0_type, wnn_limits[webnn_op_type]["a"]["dataTypes"])) {
-    LOGS(logger, VERBOSE) << "[" << op_type
-                          << "] Input type: [" << input0_type
-                          << "] is not supported for now";
-    return false;
+  std::vector<int32_t> input_types = {input0_type, input1_type};
```

Comment:
Since we know the vector has an upper limit of 4 elements, can we use `InlinedVector` rather than allocating on the heap?

```
InlinedVector<int32_t, 4> input_types = {input0_type, input1_type};
...
input_types.push_back(input2_type);
```

Reply:
Sure, done.